### PR TITLE
ANA-688: SimpleStatic pricing of various OTC instruments

### PR DESCRIPTION
**Pull request by jz-finbourne**
# Pull Request Checklist

- [ ] Read the [contributing guidelines](../blob/master/docs/CONTRIBUTING.md)
- [ ] Tests pass
- [ ] Raised the PR against the `develop` branch

# Description of the PR

Describe the code changes for the reviewers, explain the solution you have provided and how it fixes the issue

### CODE REVIEW

File: sdk/Lusid.Sdk.Tests/Utilities/TestDataUtilities.cs [hunk 1]
```diff
@@ -499,7 +499,15 @@ public static UpsertComplexMarketDataRequest ConstantVolSurfaceRequest(DateTimeO
         /// </summary>
         public static UpsertRecipeRequest BuildRecipeRequest(string recipeCode, string scope, ModelSelection.ModelEnum model)
         {
-            var pricingOptions = new PricingOptions(new ModelSelection(ModelSelection.LibraryEnum.Lusid, model));
+            // Note that inside CreatePortfolioAndInstrument, the method TestDataUtilities.BuildInstrumentUpsertRequest books the instrument using "ClientInternal".
+            // As such the quote upserted using "ClientInternal". The market rule key needs to be "ClientInternal" also to find the quote.  
+            var simpleStaticLuidRule = new MarketDataKeyRule(
```
Comment (fightingsleep) [inline sdk/Lusid.Sdk.Tests/Utilities/TestDataUtilities.cs:504]: Not sure I understand this variable name

File: sdk/Lusid.Sdk.Tests/tutorials/Instruments/ExoticExamples.cs [hunk 2]
```diff
@@ -64,5 +64,12 @@ public void ExoticCreationAndUpsertionExample()
             Assert.That(roundTripExotic.Content, Is.EqualTo(exotic.Content));
             Assert.That(roundTripExotic.InstrumentFormat, Is.EqualTo(exotic.InstrumentFormat)); 
         }
+        
+        [TestCase(ModelSelection.ModelEnum.SimpleStatic)]
+        public void InterestRateSwapValuationExample(ModelSelection.ModelEnum model)
+        {
+            var exotic = InstrumentExamples.CreateExampleExotic();
```
Comment (fightingsleep) [inline sdk/Lusid.Sdk.Tests/tutorials/Instruments/ExoticExamples.cs:71]: The function name implies valuing an interest rate swap, but you instantiate an exotic. Is this intentional? If so, add a comment explaining

File: sdk/Lusid.Sdk.Tests/Utilities/TestDataUtilities.cs (hunk 1 above)
Comment (jz-finbourne) [reply sdk/Lusid.Sdk.Tests/Utilities/TestDataUtilities.cs]: Ah yes, was when I trying to make it work. Have dropped "luid" in name, should be better. thanks

File: sdk/Lusid.Sdk.Tests/tutorials/Instruments/InterestRateSwapWithNamedConventionsExamples.cs [hunk 3]
```diff
@@ -60,5 +60,12 @@ public void InterestRateSwapWithNamedConventionsCreationAndUpsertionExample()
             // Delete Instrument
             _instrumentsApi.DeleteInstrument("ClientInternal", uniqueId); 
         }
+        
+        [TestCase(ModelSelection.ModelEnum.SimpleStatic)]
+        public void InterestRateSwapValuationExample(ModelSelection.ModelEnum model)
```
Comment (fightingsleep) [inline sdk/Lusid.Sdk.Tests/tutorials/Instruments/InterestRateSwapWithNamedConventionsExamples.cs:73]: How is this function different than the function you added to InterestRateSwapExamples.cs?

Comment (jz-finbourne) [reply sdk/Lusid.Sdk.Tests/tutorials/Instruments/InterestRateSwapWithNamedConventionsExamples.cs:73]: Copy, paste and forgot to edit - whoops!

File: sdk/Lusid.Sdk.Tests/tutorials/Instruments/InterestRateSwapExamples.cs [hunk 4]
```diff
@@ -53,5 +59,14 @@ public void InterestRateSwapCreationAndUpsertionExample()
             // Delete Instrument
             _instrumentsApi.DeleteInstrument("ClientInternal", uniqueId); 
         }
+        
+        [TestCase(ModelSelection.ModelEnum.SimpleStatic)]
+        [TestCase(ModelSelection.ModelEnum.ConstantTimeValueOfMoney)]
+        [TestCase(ModelSelection.ModelEnum.Discounting)]
+        public void InterestRateSwapValuationExample(ModelSelection.ModelEnum model)
```
Comment (fightingsleep) [inline sdk/Lusid.Sdk.Tests/tutorials/Instruments/InterestRateSwapExamples.cs:66]: This test and the one in InterestRateSwapWIthNamedConventionsExamples.cs still look identical. What is the difference?

Comment (jz-finbourne) [reply sdk/Lusid.Sdk.Tests/tutorials/Instruments/InterestRateSwapExamples.cs:72]: Yes, just pushed now. I pushed a different commit with more models first, sorry.

The diff is two different ways of creating a swap. I changed the method call inside now.

**Review by fightingsleep — APPROVED** [review]

**Pull request merged**